### PR TITLE
Add support for LLM template

### PR DESCRIPTION
**Pull request by schustmi**
## Describe changes
Adds support for the new LLM Finetuning template.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [ ] If my changes require changes to the dashboard, these changes are communicated/requested.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Comment by schustmi** [comment]
@avishniakov The CI ran (successfully) but didn't actually commit the files for the first time. Is that intended and I need to commit the initial version myself?

**Comment by avishniakov** [comment]
> @avishniakov The CI ran (successfully) but didn't actually commit the files for the first time. Is that intended and I need to commit the initial version myself?

No actually, it should always commit changes anyway

File: .typos.toml [hunk 1]
```diff
@@ -7,6 +7,7 @@ extend-exclude = [
     "tests/unit/materializers/test_built_in_materializer.py",
     "tests/integration/functional/cli/test_pipeline.py",
     "src/zenml/zen_server/dashboard/",
+    "examples/llm_finetuning/lit_gpt/"
```
Comment (avishniakov) [inline .typos.toml:10]: Why do you need to exclude the new template from the typos check? Can the typos be fixed on the template side instead?

File: pyproject.toml [hunk 2]
```diff
@@ -303,6 +303,12 @@ exclude = [
     "venv",
     '__init__.py',
     'src/zenml/cli/version.py',
+    # LitGPT files from the LLM Finetuning example
+    'examples/llm_finetuning/evaluate',
+    'examples/llm_finetuning/finetune',
+    'examples/llm_finetuning/generate',
+    'examples/llm_finetuning/lit_gpt',
+    'examples/llm_finetuning/scripts',
```
Comment (avishniakov) [inline pyproject.toml:311]: Also not very confident here - why is this needed? You have linting in your `copier.yaml`, right? Why this is not aligned with what we have here?

**Review by avishniakov — COMMENTED** [review]
Left a few questions. Can you also add the image optimizer to the template repo, so optimization is not happening here and will not be causing infinite updates?
https://github.com/zenml-io/template-starter/blob/main/.github/workflows/image-optimizer.yml

File: .typos.toml (hunk 1 above)
Comment (schustmi) [reply .typos.toml:10]: The template contains third-party code which I tried not to modify, as we'll have to swap that from time to time to upgrade it.

File: pyproject.toml (hunk 2 above)
Comment (schustmi) [reply pyproject.toml:311]: Same as above.

**Comment by schustmi** [comment]
@avishniakov Regarding the image optimizer: That's already in the repo, but probably didn't run for this PR yet. What would be the best solution to resolve this? Push the optimized image to the other repo and re-tag?

**Comment by avishniakov** [comment]
> @avishniakov Regarding the image optimizer: That's already in the repo, but probably didn't run for this PR yet. What would be the best solution to resolve this? Push the optimized image to the other repo and re-tag?

It would the easiest way, yeah. Let's do it like this. In this case - this PR can go directly!

**Comment by schustmi** [comment]
It seems like it's actually just the same image 🤔
I'll re-run the job here to see if it still detects changes
